pbf/api.py

```diff
@@ -73,18 +75,34 @@ def get_ip(request):
     else:
         return request.META["REMOTE_ADDR"]
 
+class ApiKey(APIKeyHeader):
+    param_name = "X-API-Key"
+    def authenticate(self, request, key):
+        secret = os.getenv('CUSTOM_API_KEY', None)
+        if secret is None:
+            return False
+        if key == secret:
+            return key
+
+header_key = ApiKey()
+
 def ip_whitelist(request):
-    ip = str(os.environ['OWN_IP'])
-    if ip is None:
-        ip = "127.0.0.1"
-    if get_ip(request) == ip:
-        return ip
+    ALLOWED_IPS = (os.getenv('ALLOWED_IPS') or "10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,127.0.0.1").split(",")
```

Comment:
I think you shouldn't feel obligated to allow all the RFC 1918 ranges by default and the default can remain at only 127.0.0.1. I think it's good for the person configuring ALLOWED_IPS to be intentional and thoughtful about what they're allowing.